<launch>
  <arg name="waypoint_set" default="aaf_waypoints"/>
  <arg name="meta_name" default="waypoint_set"/>
  <arg name="collection_name" default="aaf_walking_group"/>
  <arg name="music_folder" default="$(find music_player)/media"/>
  <arg name="waypoint_sounds_folder" default="$(find aaf_waypoint_sounds)/media"/>
  <arg name="waypoint_sounds_file" default="$(find aaf_walking_group)/conf/waypoint_sounds.yaml" />
  <arg name="music_set_group" default="aaf_walking_group"/>
  <arg name="music_set_waypoints" default="aaf_waypoint_sounds"/>
  <arg name="music_set_jingles" default="aaf_jingles"/>
  <arg name="video_set" default="aaf_video_set"/>
  <arg name="image_set" default="walking_group_pictures"/>

  <node pkg="aaf_walking_group" type="wait_for_participant.py" name="wait_for_participant" output="screen" respawn="true"/>
  <node pkg="aaf_walking_group" type="guiding_action.py" name="guiding_server" output="screen" respawn="true"/>
  <node pkg="aaf_walking_group" type="media_server.py" name="image_server" output="screen" respawn="true">
    <param name="media_set" type="string" value="$(arg image_set)"/>
    <param name="media_set_type" type="string" value="Photo"/>
  </node>
  <node pkg="aaf_walking_group" type="media_server.py" name="video_server" output="screen" respawn="true">
    <param name="media_set" type="string" value="$(arg video_set)"/>
    <param name="media_set_type" type="string" value="Video"/>
  </node>
  <node pkg="social_card_reader" type="social_card_reader" name="social_card_reader" output="screen" respawn="true"/>
  <node pkg="aaf_walking_group" type="interface_server.py" name="interface_server" output="screen" respawn="true"/>
  <node pkg="aaf_walking_group" type="walking_gui_server.py" name="walking_interface_server" output="screen" respawn="true"/>
  <node pkg="aaf_walking_group" type="positionOfCard.py" name="QSR_generator" output="screen" respawn="true"/>
  <node pkg="aaf_walking_group" type="aaf_walking_group_state_machine.py" name="walking_group_smach" output="screen" respawn="true">
    <param name="mongodb_params/waypointset_name" type="string" value="$(arg waypoint_set)"/>
    <param name="mongodb_params/waypointset_collection" type="string" value="$(arg collection_name)"/>
    <param name="mongodb_params/waypointset_meta" type="string" value="$(arg meta_name)"/>
  </node>

  <include file="$(find aaf_waypoint_sounds)/launch/waypoint_sounds.launch">
    <arg name="music_set" value="$(arg music_set_waypoints)"/>
    <arg name="waypoint_sounds_file" value="$(arg waypoint_sounds_file)"/>
  </include>
  <node pkg="music_player" type="music_player_server.py" name="music_player_server" respawn="true">
    <param name="music_set" type="string" value="$(arg music_set_group)"/>
  </node>
  <include file="$(find sound_player_server)/launch/sound_player.launch">
    <arg name="music_set" value="$(arg music_set_jingles)"/>
  </include>
  <node pkg="walking_group_recovery" type="toggle_walking_group_recovery_server.py" name="toggle_walking_group_recovery_server" respawn="true"/>

</launch>
